ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -0,0 +1,272 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtQuick.Shapes 1.15
+
+import StatusQ 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Utils 0.1 as SQUtils
+import StatusQ.Core.Theme 0.1
+
+import shared.popups.send.views 1.0
+import shared.popups.send.panels 1.0
+
+import utils 1.0
+import shared.stores 1.0
+
+import SortFilterProxyModel 0.2
+
+Control {
+    id: root
+
+    // input API
+    required property CurrenciesStore currencyStore
```

Comment:
I would like to remove stores from here completely. Seems like we only need currentCurrency and formatCurrencyAMount functions so we can add those as functions and properties?

Reply:
We could... but (just personal opinion) I hate having JS functions as part of the component API
(sth like:
```qml
Item {
    id: root

    property var assetsModel
    property var collectiblesModel
    property var networksModel
    property bool onlyAssets: true
    property string searchText

    property var formatCurrentCurrencyAmount: function(balance){}
    property var formatCurrencyAmountFromBigInt: function(balance, symbol, decimals){}

```

Opinions @alexjba @micieslak @noeliaSD ?

Reply:
I'd opt for the store for now because the HoldingsSelector already expects this function as input. But we could look for alternatives later when refactoring the HoldingsSelector.

Reply:
Another case in point:
```qml
required property var formatCurrentCurrencyAmount: function(balance){}
```
if we want to have a nice and strict API, the above is of course impossible (would trigger an error)